meteomatics/src/util.rs

```diff
@@ -0,0 +1,399 @@
+    pub stats: UserStats,
+}
+
+/// This contains the actual information about the account statistics, quota and permissions.
+#[derive(Debug, Deserialize, Serialize)]
+pub struct UserStats{
+    pub username: String,
+    #[serde(rename(serialize = "requests total", deserialize = "requests total"))]
+    pub total: Limit,
+    #[serde(rename(serialize = "requests since last UTC midnight", deserialize = "requests since last UTC midnight"))]
+    pub since_midnight: Limit,
+    #[serde(rename(serialize = "requests since HH:00:00", deserialize = "requests since HH:00:00"))]
+    pub since_0: Limit,
+    #[serde(rename(serialize = "requests in the last 60 seconds", deserialize = "requests in the last 60 seconds"))]
+    pub since_60s: Limit,
+    #[serde(rename(serialize = "requests in parallel", deserialize = "requests in parallel"))]
+    pub parallel: Limit,
+    #[serde(rename(serialize = "historic request option", deserialize = "historic request option"))]
+    pub hist: String,
+    #[serde(rename(serialize = "area request option", deserialize = "area request option"))]
+    pub area: bool,
+    #[serde(rename(serialize = "model select option", deserialize = "model select option"))]
+    pub models: Vec<String>,
+    #[serde(rename(serialize = "error message", deserialize = "error message"))]
+    pub error: String,
+    #[serde(rename(serialize = "contact emails", deserialize = "contact emails"))]
+    pub contact: Vec<String>,
+}
+
+/// The Limit struct is used to de-serialize the limit attributes of the account (e.g. how many 
+/// requests in parallel are allowed etc.)
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Limit{
+    pub used: u32,
+    #[serde(rename(serialize = "soft limit", deserialize = "soft limit"))]
+    pub soft_lim: u32,
+    #[serde(rename(serialize = "hard limit", deserialize = "hard limit"))]
+    pub hard_lim: u32
+}
+
+// Deserializes the response for the user_stats_json query.
+pub async fn extract_user_statistics(response: Response) -> std::result::Result<UStatsResponse, ConnectorError> {
+    let json: UStatsResponse = response.json::<UStatsResponse>().await?;
+    Ok(json)
+}
+
+/// Writes the HTTP response to a file. 
+pub async fn write_file(response: Response, file_name: &String) -> std::result::Result<(), ConnectorError> {
+    let body = response.bytes().await?;
+    let mut content = std::io::Cursor::new(body);
+
+    let mut file = File::create(file_name)?;
+    std::io::copy(&mut content, &mut file)?;
+    Ok(())
+}
+
+/// Creates a path if it does not already exist.
+pub async fn create_path(file_name: &String) -> std::result::Result<(), ConnectorError> {
+    // https://www.programming-idioms.org/idiom/212/check-if-folder-exists
+    let dir: &Path = Path::new(file_name).parent().unwrap();
+    let b: bool = dir.is_dir();
+
+    if !b {
+       match fs::create_dir_all(dir) {
+           Ok(_) => Ok(()),
+           Err(_) => Err(ConnectorError::FileIOError)
+       }
+    } else {
```

Comment:
Probably you do not need this else, since you match and return Ok or Error, so the code below will anyway not be hit.